### ISSUE TITLE
Firestore futures warning after updating to 6.15.0

### ISSUE THREAD

**Issue by shekit** [firebase/quickstart-unity#723]
### Please fill in the following fields:
Unity editor version: 2019.3.11f1
Firebase Unity SDK version: 6.15.0
Source you installed the SDK (.unitypackage or Unity Package Manager): UPM
Firebase plugins in use (Auth, Database, etc.): Auth, Firestore, Storage
Additional SDKs you are using (Facebook, AdMob, etc.):  NA
Platform you are using the Unity editor on (Mac, Windows, or Linux):  Mac
Platform you are targeting (iOS, Android, and/or desktop): iOS/Android
Scripting Runtime (Mono, and/or IL2CPP): IL2CPP

### Please describe the issue here:
(Please list the full steps to reproduce the issue. Include device logs, Unity logs, and stack traces if available.)

Since upgrading to 6.15.0 from 6.14.0 I have begun receiving these warnings in the console. I'm including a couple of stack traces below though I seem to be receiving a warning for every call to a firebase sdk

```
Future with handle 1 still exists though its backing API 0xDC3ECCA0 is being deleted. Please call Future::Release() before deleting the backing API.
UnityEngine.Debug:LogWarning(Object)
Firebase.Platform.FirebaseLogger:LogMessage(PlatformLogLevel, String) (at Z:/tmp/tmp.la3ltWQdKA/firebase/app/client/unity/src/Unity/FirebaseLogger.cs:92)
Firebase.LogUtil:LogMessage(LogLevel, String) (at Z:/tmp/tmp.txs8ldQ514/firebase/app/client/unity/proxy/LogUtil.cs:68)
Firebase.LogUtil:LogMessageFromCallback(LogLevel, String) (at Z:/tmp/tmp.txs8ldQ514/firebase/app/client/unity/proxy/LogUtil.cs:76)
Firebase.AppUtilPINVOKE:PollCallbacks()
Firebase.AppUtil:PollCallbacks() (at Z:/tmp/tmp.txs8ldQ514/firebase/app/client/unity/proxy/AppUtil.cs:32)
Firebase.Platform.FirebaseAppUtils:PollCallbacks() (at Z:/tmp/tmp.txs8ldQ514/firebase/app/client/unity/proxy/FirebaseAppUtils.cs:33)
Firebase.Platform.FirebaseHandler:Update() (at Z:/tmp/tmp.la3ltWQdKA/firebase/app/client/unity/src/Unity/FirebaseHandler.cs:205)
Firebase.Platform.FirebaseMonoBehaviour:Update() (at Z:/tmp/tmp.la3ltWQdKA/firebase/app/client/unity/src/Unity/FirebaseMonoBehaviour.cs:45)

```

```
Future with handle 2 still exists though its backing API 0xDF82E710 is being deleted. Please call Future::Release() before deleting the backing API.
UnityEngine.Debug:LogWarning(Object)
Firebase.Platform.FirebaseLogger:LogMessage(PlatformLogLevel, String) (at Z:/tmp/tmp.la3ltWQdKA/firebase/app/client/unity/src/Unity/FirebaseLogger.cs:92)
Firebase.LogUtil:LogMessage(LogLevel, String) (at Z:/tmp/tmp.txs8ldQ514/firebase/app/client/unity/proxy/LogUtil.cs:68)
Firebase.LogUtil:LogMessageFromCallback(LogLevel, String) (at Z:/tmp/tmp.txs8ldQ514/firebase/app/client/unity/proxy/LogUtil.cs:76)
Firebase.Storage.StorageInternalPINVOKE:StorageReferenceInternal_GetDownloadUrl(HandleRef)
Firebase.Storage.StorageReferenceInternal:GetDownloadUrlAsync() (at Z:/tmp/tmp.38nope6H9v/firebase/storage/client/unity/proxy/StorageReferenceInternal.cs:64)
Firebase.Storage.StorageReference:GetDownloadUrlAsync() (at Z:/tmp/tmp.38nope6H9v/firebase/storage/client/unity/proxy/StorageReference.cs:497)
<GetGameImageURL>d__7:MoveNext() (at Assets/Scripts/UI/FirestoreManager.cs:64)
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1:Start(<GetGameImageURL>d__7&)
FirestoreManager:GetGameImageURL(GameData)
<CreateGamePrefab>d__16:MoveNext() (at Assets/Scripts/UI/flow/ProfileUI.cs:118)
System.Runtime.CompilerServices.AsyncVoidMethodBuilder:Start(<CreateGamePrefab>d__16&)
ProfileUI:CreateGamePrefab(GameData)
ProfileUI:SetupUI() (at Assets/Scripts/UI/flow/ProfileUI.cs:101)
<LoadData>d__13:MoveNext() (at Assets/Scripts/UI/flow/ProfileUI.cs:72)
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1:SetResult(List`1)
<GetUserGames>d__9:MoveNext() (at Assets/Scripts/UI/FirestoreManager.cs:128)
System.Threading.Tasks.TaskCompletionSource`1:SetResult(Boolean)
Firebase.<RunAsync>c__AnonStorey1`1:<>m__0() (at Z:/tmp/tmp.la3ltWQdKA/firebase/app/client/unity/src/Platform/Dispatcher.cs:77)
Firebase.ExceptionAggregator:Wrap(Action) (at Z:/tmp/tmp.la3ltWQdKA/firebase/app/client/unity/src/Platform/ExceptionAggregator.cs:112)
Firebase.Dispatcher:PollJobs() (at Z:/tmp/tmp.la3ltWQdKA/firebase/app/client/unity/src/Platform/Dispatcher.cs:123)
Firebase.Platform.FirebaseHandler:Update() (at Z:/tmp/tmp.la3ltWQdKA/firebase/app/client/unity/src/Unity/FirebaseHandler.cs:204)
Firebase.Platform.FirebaseMonoBehaviour:Update() (at Z:/tmp/tmp.la3ltWQdKA/firebase/app/client/unity/src/Unity/FirebaseMonoBehaviour.cs:45)

```

On reverting to 6.14.1, the warnings no longer appear

### Please answer the following, if applicable:
Have you been able to reproduce this issue with just the Firebase Unity quickstarts (this GitHub project)? NA

What's the issue repro rate? (eg 100%, 1/5 etc) NA

**Comment by EgilSandfeld** [comment]
Getting the exact same messages in Xcode log for my iOS app, running Unity 2019.3.15f1 with Firebase 6.15.0.

Possibly related issues [here] on Unity forum(https://forum.unity.com/threads/unityframework-load-in-2019-3-11-causes-watchdog-transgression-crash-0x8badf00d.878620/)

**Comment by shekit** [comment]
@EgilSandfeld I'm actually facing the crash as well. Looking at the device logs it does point to Springboard. I'm facing this issue with 6.14.0 and unity 2019.3.11f1. 

The weird part is, when I open the app it crashes. However if I immediately relaunch the app again, it opens without issues. I have tried all the unity recommendation of adding in a simple loader scene and delaying load by a frame or two but the issue still persists.

**Comment by alexames** [comment]
Hi folks, sorry for the late reply. 

The error messages you were seeing should be fixed if you update to the latest release (6.15.2). 

@shekit I don't think the crash you're reporting here is related to the error messages, but I could be wrong. Can you upgrade to the latest version, and if the issue is still occuring open a new issue.
